src/components/FileList.tsx

```diff
@@ -110,93 +107,119 @@ export class FileList extends React.Component<IFileListProps, IFileListState> {
       disableFiles: false
     };
 
-    /** Add right-click menu options for files in repo
-     *
-     */
-
-    if (!commands.hasCommand(CommandIDs.gitFileOpen)) {
-      commands.addCommand(CommandIDs.gitFileOpen, {
-        label: 'Open',
-        caption: 'Open selected file',
-        execute: () => {
-          try {
-            this.openListedFile(
-              this.state.contextMenuTypeX,
-              this.state.contextMenuTypeY,
-              this.state.contextMenuFile,
-              this.props.app
-            );
-          } catch (err) {}
-        }
-      });
-    }
-    if (!commands.hasCommand(CommandIDs.gitFileStage)) {
-      commands.addCommand(CommandIDs.gitFileStage, {
-        label: 'Stage',
-        caption: 'Stage the changes of selected file',
-        execute: () => {
-          try {
-            this.addUnstagedFile(
-              this.state.contextMenuFile,
-              this.props.topRepoPath,
-              this.props.refresh
-            );
-          } catch (err) {}
-        }
-      });
-    }
-    if (!commands.hasCommand(CommandIDs.gitFileTrack)) {
-      commands.addCommand(CommandIDs.gitFileTrack, {
-        label: 'Track',
-        caption: 'Start tracking selected file',
-        execute: () => {
-          try {
-            this.addUntrackedFile(
+    commands.addCommand(CommandIDs.gitFileOpen, {
+      label: 'Open',
+      caption: 'Open selected file',
+      execute: async () => {
+        try {
+          await this.openListedFile(
+            this.state.contextMenuTypeX,
+            this.state.contextMenuTypeY,
+            this.state.contextMenuFile,
+            this.props.app
+          );
+        } catch (err) {}
```

Comment:
Would you at least want to log an error here?

Reply:
This try catch should'nt been there in the first place. Removed.